t/admin/upstream4.t

```diff
@@ -560,3 +560,35 @@ passed
     }
 --- response_body
 [delete] code: 200 message: passed
+
+
+
+=== TEST 17: patch upstream with sub_path, the data is number
+--- config
+    location /t {
+        content_by_lua_block {
+            local json = require("toolkit.json")
+            local t = require("lib.test_admin").test
+            local code, message = t('/apisix/admin/upstreams/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "nodes": {},
+                    "type": "roundrobin"
+                 }]]
+            )
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(message)
+                return
+            end
+
+            local code, message = t('/apisix/admin/upstreams/1/retries',
+                 ngx.HTTP_PATCH,
+                 json.encode(1)
+            )
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(message)
```

Comment:
I suggest add PATCH body is json format like:
```
curl http://127.0.0.1:9080/apisix/admin/upstreams/100/nodes  -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -i -X PATCH -d '
{
    "39.97.63.215:80": 1,
    "39.97.63.215:81": 1
}'
```

Reply:
PATCH with json has been covered by https://github.com/apache/apisix/blob/183351cd9f68df706d467f4474ca77eaec398100/t/admin/upstream3.t#L348

Reply:
> PATCH with json has been covered by
> 
> https://github.com/apache/apisix/blob/183351cd9f68df706d467f4474ca77eaec398100/t/admin/upstream3.t#L348

we have auto test cases, why we don't find this issue ?

Reply:
This case is not covered by the test case.